crates/full-node/sov-sequencer/src/preferred/async_batch.rs

```diff
@@ -66,11 +67,28 @@ impl<S: Spec> MaybeAsyncBatch<S> {
 /// The control flow injector for an async batch.
 #[derive(Debug)]
 pub enum MaybeAsyncBatchControlFlow<S: Spec> {
-    Async(AsyncBatchResponder<S>),
+    Async {
+        responder: AsyncBatchResponder<S>,
+        // If this is the main sequencer executor, it may benefit from using a precomputed
+        // transaction change set from the worker warm-up executor.
+        maybe_tx_change_set: Option<oneshot::Receiver<TxChangeSet>>,
```

Comment:
non-blocking, nit: Maybe we could rename this to `tx_read_set_hint`, since it's supposed to be the set of items that were read, not the items that were changed.